game-core/src/main/java/games/strategy/util/OptionalUtils.java

```diff
@@ -11,6 +12,33 @@
 public final class OptionalUtils {
   private OptionalUtils() {}
 
+  /**
+   * If a value is present in both {@code optional1} and {@code optional2}, performs the given action with the values,
+   * otherwise does nothing.
+   */
+  public static <T, U> void ifAllPresent(
```

Comment:
Yes, I envisioned more overloads being added for three or more `Optional`s to ensure type safety.

I'm pretty sure I saw another location in the code where we effectively check for the presence of three `Optional`s before doing something.  In that case, there is no suppletive variant for "all three", so we'd have `ifAllThreePresent()`, which seems redundant.  Therefore, I think keeping the overloads for different `Optional` parameter counts named consistently would be preferable so a dev doesn't have to change the method name at the call site simply because they add an additional argument.

Reply:
Thanks, considering the overload route, I'd agree to keeping the current name.